<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>HomepageUi</class>
 <widget class="QWidget" name="HomepageUi">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>720</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1100</width>
    <height>720</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1100</width>
    <height>720</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="styleSheet">
   <string notr="true">HomePageUi {
background-color: #E0E0E0;
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3" stretch="1,1">
   <property name="spacing">
    <number>10</number>
   </property>
   <property name="leftMargin">
    <number>20</number>
   </property>
   <property name="topMargin">
    <number>10</number>
   </property>
   <property name="rightMargin">
    <number>20</number>
   </property>
   <property name="bottomMargin">
    <number>14</number>
   </property>
   <item>
    <widget class="QFrame" name="frame_status">
     <property name="styleSheet">
      <string notr="true">QFrame#frame_status {
border-radius: 10px 10px 10px 10px;
border-image: url(:/ui/images/bg_ucar.png) 0 0 0 0 stretch stretch;
}
</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4" stretch="40,2,5,2,5">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="label_2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>350</width>
          <height>38</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">QLabel {
background-color: transparent;
color: #FFFFFF;
font-family: SourceHanSansCN-Medium;
font-size: 28px;
text-align: center;
}</string>
        </property>
        <property name="text">
         <string>RoboGo软件工具</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>40</number>
        </property>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <property name="spacing">
           <number>10</number>
          </property>
          <item alignment="Qt::AlignVCenter">
           <widget class="QLabel" name="lb_status_icon">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>16</width>
              <height>16</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16</width>
              <height>16</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">QLabel[on_line=true] {
background-color: #14DE32;
border-radius: 8px;
}
QLabel[on_line=false] {
background-color: #ED3030;
border-radius: 8px;
}</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="scaledContents">
             <bool>true</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
            <property name="on_line" stdset="0">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item alignment="Qt::AlignVCenter">
           <widget class="QLabel" name="lb_status">
            <property name="styleSheet">
             <string notr="true">QLabel {
background-color: transparent;
color: #FFFFFF;
font-family: SourceHanSansCN-Normal;
font-size: 14px;
text-align: center;
}</string>
            </property>
            <property name="text">
             <string>离线</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item alignment="Qt::AlignHCenter">
         <widget class="QLabel" name="label_6">
          <property name="styleSheet">
           <string notr="true">QLabel {
background-color: transparent;
color: #FFFFFF;
font-family: SourceHanSansCN-Normal;
font-size: 14px;
text-align: center;
}</string>
          </property>
          <property name="text">
           <string>波特率: 115200</string>
          </property>
         </widget>
        </item>
        <item alignment="Qt::AlignHCenter">
         <widget class="QPushButton" name="pb_guide">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>150</width>
            <height>34</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>150</width>
            <height>34</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">QPushButton {
background-color: #007AFF;
color: #FFFFFF;
font-family: SourceHanSansCN-Normal;
font-size: 14px;
text-align: center;
border-radius: 15px;
}</string>
          </property>
          <property name="text">
           <string>连接指引</string>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="2,1">
     <property name="spacing">
      <number>10</number>
     </property>
     <item>
      <widget class="QFrame" name="frame_3">
       <property name="styleSheet">
        <string notr="true">.QFrame {
border-radius: 10px 10px 10px 10px;
background-color: #FFFFFF;
border: 1px solid #C7C7C7;
}
</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <property name="spacing">
         <number>40</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>56</number>
          </property>
          <property name="topMargin">
           <number>55</number>
          </property>
          <property name="bottomMargin">
           <number>55</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item alignment="Qt::AlignHCenter">
             <widget class="QLabel" name="label_3">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>91</width>
                <height>24</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>91</width>
                <height>24</height>
               </size>
              </property>
              <property name="styleSheet">
               <string notr="true">QLabel {
background-color: transparent;
color: #333333;
font-family: SourceHanSansCN-Normal;
font-size: 14px;
text-align: center;
}</string>
              </property>
              <property name="lineWidth">
               <number>0</number>
              </property>
              <property name="text">
               <string>角度X:</string>
              </property>
             </widget>
            </item>
            <item alignment="Qt::AlignHCenter">
             <widget class="QLabel" name="lb_angle_x_value">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>180</width>
                <height>40</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>180</width>
                <height>40</height>
               </size>
              </property>
              <property name="styleSheet">
               <string notr="true">QLabel {
background-color: #101010;
color: #0CF42D;
font-family: SourceHanSansCN-Medium;
font-size: 30px;
text-align: center;
border-radius:6px;
}</string>
              </property>
              <property name="text">
               <string>0°</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item alignment="Qt::AlignHCenter">
               <widget class="QLabel" name="label_5">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>91</width>
                  <height>24</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>91</width>
                  <height>24</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">QLabel {
background-color: transparent;
color: #333333;
font-family: SourceHanSansCN-Normal;
font-size: 14px;
text-align: center;
}</string>
                </property>
                <property name="lineWidth">
                 <number>0</number>
                </property>
                <property name="text">
                 <string>角度Y:</string>
                </property>
               </widget>
              </item>
              <item alignment="Qt::AlignHCenter">
               <widget class="QLabel" name="lb_angle_y_value">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>180</width>
                  <height>40</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>180</width>
                  <height>40</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">QLabel {
background-color: #101010;
color: #0CF42D;
font-family: SourceHanSansCN-Medium;
font-size: 30px;
text-align: center;
border-radius:6px;
}</string>
                </property>
                <property name="text">
                 <string>0°</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item alignment="Qt::AlignHCenter">
             <widget class="QLabel" name="label_7">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>91</width>
                <height>24</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>91</width>
                <height>24</height>
               </size>
              </property>
              <property name="styleSheet">
               <string notr="true">QLabel {
background-color: transparent;
color: #333333;
font-family: SourceHanSansCN-Normal;
font-size: 14px;
text-align: center;
}</string>
              </property>
              <property name="text">
               <string>角度Z:</string>
              </property>
             </widget>
            </item>
            <item alignment="Qt::AlignHCenter">
             <widget class="QLabel" name="lb_angle_z_value">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>180</width>
                <height>40</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>180</width>
                <height>40</height>
               </size>
              </property>
              <property name="styleSheet">
               <string notr="true">QLabel {
background-color: #101010;
color: #0CF42D;
font-family: SourceHanSansCN-Medium;
font-size: 30px;
text-align: center;
border-radius:6px;
}</string>
              </property>
              <property name="text">
               <string>0°</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_11">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>35</number>
          </property>
          <property name="topMargin">
           <number>35</number>
          </property>
          <property name="rightMargin">
           <number>35</number>
          </property>
          <property name="bottomMargin">
           <number>35</number>
          </property>
          <item>
           <widget class="Gradienter" name="gradienter" native="true">
            <property name="minimumSize">
             <size>
              <width>260</width>
              <height>260</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>260</width>
              <height>260</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="frame_compass">
       <property name="styleSheet">
        <string notr="true">.QFrame {
border-radius: 10px 10px 10px 10px;
background-color: #FFFFFF;
border: 1px solid #C7C7C7;
}
</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <widget class="QFrame" name="frame">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>260</width>
            <height>260</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>260</width>
            <height>260</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">QFrame#frame{
border-radius: 10px 10px 10px 10px;
border-image: url(:/ui/images/compass_bg.png) 0 0 0 0 stretch stretch;
}
</string>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <property name="lineWidth">
           <number>0</number>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <property name="spacing">
            <number>0</number>
           </property>
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="lb_compass_needle">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>220</width>
               <height>220</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>220</width>
               <height>220</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="pixmap">
              <pixmap resource="../resource/resource.qrc">:/ui/images/compass_needle.png</pixmap>
             </property>
             <property name="scaledContents">
              <bool>false</bool>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Gradienter</class>
   <extends>QWidget</extends>
   <header>gradienter.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resource/resource.qrc"/>
 </resources>
 <connections/>
</ui>
